apiAssignment/src/ApiCall.java

```diff
@@ -0,0 +1,127 @@
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+
+public class ApiCall
+{
+    public static void main(String[] args) throws Exception
+    {
+        Worker w1=new Worker("https://www.boredapi.com/api/activity?participants=1");
+        Worker w2=new Worker("https://randomuser.me/api/");
+        Worker w3=new Worker("https://baconipsum.com/api/?type=all-meat&paras=3&start-with-lorem=1&format=json");
+        ExecutorService pool= Executors.newFixedThreadPool(3);
+        Future<String>w1Return=pool.submit(w1);
+        Future<String>w2Return=pool.submit(w2);
+        Future<String>w3Return=pool.submit(w3);
+        System.out.println(w2Return.get()+"\n\n"+w1Return.get()+"\n\n"+w3Return.get());
```

Comment:
Will it wait for all threads to be completed?

Reply:
yes it will wait and store the result in future object then it will read that future object